misk-hibernate/src/main/kotlin/misk/hibernate/ReflectionQueryFactory.kt

```diff
@@ -252,7 +259,7 @@ internal class ReflectionQuery<T : DbEntity<T>>(
 
     query.orderBy(buildOrderBys(queryRoot, criteriaBuilder))
 
-    val typedQuery = session.hibernateSession.createQuery(query)
+    val typedQuery = createTypedQueryWithHint(session, query)
```

Comment:
nit: the `withHint` is misleading, makes it sound like you always add a hint. Could just drop it from the name.